plugins/modules/nd_api_key.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Dev Sinha (@DevSinha13) <devsinh@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: nd_api_key
+version_added: "1.4.0"
+short_description: Manage API keys in Nexus Dashboard
+description:
+- Manage API keys in Cisco Nexus Dashboard (ND).
+- It supports creating, updating, querying, and deleting API keys.
+author:
+- Dev Sinha (@DevSinha13)
+options:
+  api_key_id:
+    description:
+    - The ID of the API key.
+    type: str
+  api_key_name:
+    description:
+    - The name of the API key.
+    - Required when state is 'present'.
+    - Must be 1-32 characters long.
+    - Only alphanumeric characters, underscores, periods, and hyphens are allowed.
+    type: str
+    aliases: [ description ]
+  annotations:
+    description:
+    - The annotations attached to the API key.
+    - Annotations are optional custom metadata to add to the API key.
+    - Stored in a key value pair in the form (key: value).
+    type: dict
+  state:
+    description:
+    - The desired state of the API key.
+    type: str
+    choices: [ present, absent, query ]
+    default: present
+extends_documentation_fragment:
+- cisco.nd.modules
+- cisco.nd.check_mode
+"""
+
+EXAMPLES = r"""
+- name: Create a new API key
+  cisco.nd.nd_api_key:
+    api_key_name: "ansible_test_key"
+    annotations:
+      owner: "ansible"
+      purpose: "automation"
+    state: present
+  register: result
+
+- name: Create API key with minimal config
+  cisco.nd.nd_api_key:
+    api_key_name: "simple_key"
+    state: present
+
+- name: Update API key annotations
+  cisco.nd.nd_api_key:
+    api_key_name: "existing_key"
+    annotations:
+      owner: "updated_owner"
+      environment: "production"
+    state: present
+
+- name: Query an existing API key by name
+  cisco.nd.nd_api_key:
+    api_key_name: "ansible_test_key"
+    state: query
+  register: query_result
+
+- name: Query all API keys
+  cisco.nd.nd_api_key:
+    state: query
+  register: all_keys
+
+- name: Delete an API key by ID
+  cisco.nd.nd_api_key:
+    api_key_id: "12345678-1234-1234-1234-1234567890ab"
+    state: absent
+
+- name: Delete an API key by name
+  cisco.nd.nd_api_key:
+    api_key_name: "ansible_test_key"
+    state: absent
+
+"""
+
+RETURN = r"""
```

Comment:
I do not think we specify RETURN in any of the other modules. How did you determine this format?